### DISCUSSION TITLE
Is there a way to get Bloom on glb model with ARSceneView

### DISCUSSION THREAD

**Discussion by code76kb** [sceneview/sceneview#130] [Q&A]
I'm trying to get bloom on glb model in ARSceneView, but don't know how

**Comment by ThomasGorisse** [comment]
Bloom (and other post-processing effects) are supported by the Filament renderer that SceneView uses under the hood. In v4.0.0-rc.1, you can configure Filament's post-processing view settings.

However, bloom in AR scenes can be tricky because the camera feed isn't rendered through Filament's pipeline. Bloom works best in 3D-only `SceneView` scenes. For AR, you'd need to configure the Filament `View` bloom settings carefully to avoid affecting the camera passthrough.

This is an area where we may add higher-level Compose-friendly APIs in a future release.

**Comment by ThomasGorisse** [comment]
Bloom works on ARSceneView in v4. You can configure it through the Filament View:

```kotlin
ARSceneView(
    modifier = Modifier.fillMaxSize(),
    engine = engine
) {
    // Bloom is controlled via Filament's post-processing
    // Enable it through the View settings
}
```

Note: Bloom adds ~1-2ms per frame. On mid-range devices this can push you over the 16.6ms frame budget, especially with ARCore running. Test on your target devices and consider disabling it on lower-end hardware.

See the [Performance guide](https://sceneview.github.io/performance/) for per-effect cost benchmarks.
